File: solutions/Move-Zeroes/Move-Zeroes.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function (nums) {
  let nonZeroIndex = 0;
  for (let index = 0; index < nums.length; index++) {
    if (nums[index] !== 0) {
      nums[nonZeroIndex++] = nums[index];
    }
  }
  while (nonZeroIndex < nums.length) {
    nums[nonZeroIndex++] = 0;
  }
};

File: solutions/Move-Zeroes/Move-Zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int nonZeroIndex = 0;
        for(int i=0;i<nums.size();i++) {
            if(nums[i] != 0) {
                nums[nonZeroIndex++] = nums[i];
            }
        }
        for(int i=nonZeroIndex;i<nums.size();i++) {
            nums[i] = 0;
        }
    }
};